js/gestionPresupuesto.js

```diff
@@ -1,18 +1,90 @@
-// TODO: Crear las funciones, objetos y variables indicadas en el enunciado
+let presupuesto = 0;
+let gastos = [];
+let idGasto = 0;
 
-// TODO: Variable global
-
-
-function actualizarPresupuesto() {
-    // TODO
+function actualizarPresupuesto(numero) {
+    if(numero>=0){
+        presupuesto = numero;
+        return numero;
+    }
+    else{
+        return -1;
+    }
 }
 
 function mostrarPresupuesto() {
-    // TODO
+    return "Tu presupuesto actual es de " + presupuesto + " €";
+}
+
+function CrearGasto(descripcion, valor, fecha, ...etiquetas) {
+
+this.valor = (valor >=0) ? valor : 0;
+this.descripcion = descripcion;
+this.etiquetas= etiquetas.length > 0 ? etiquetas:[];
```

Comment:
En lugar de esto, puedes utilizar `this.anyadirEtiqueta` para evitar crear un gasto con etiquetas duplicadas.